pandas/_libs/src/ujson/lib/ultrajsondec.c

```diff
@@ -160,12 +161,15 @@ FASTCALL_ATTR JSOBJ FASTCALL_MSVC decode_numeric(struct DecoderState *ds) {
                 // FIXME: Check for arithmetic overflow here
                 // PERF: Don't do 64-bit arithmetic here unless we know we have
                 // to
-                intValue = intValue * 10ULL + (JSLONG)(chr - 48);
+                newDigit = (JSLONG)(chr - 48);
 
-                if (intValue > overflowLimit) {
-                    return SetError(ds, -1, overflowLimit == LLONG_MAX
-                                                ? "Value is too big"
-                                                : "Value is too small");
+                if (intValue*10ULL + newDigit > overflowLimit) {
```

Comment:
I don't think this works generally. If you want a reference for overflow detection try looking at the python implementation of PyLong_AsLongLongAndOverflow

Reply:
I feel like this *might* be ok because overflowLimit is adjusted earlier on depending on whether  the number is positive or negative (in line 140 `overflowLimit` gets changed if intValue came with a minus sign):
https://github.com/pandas-dev/pandas/blob/0159cba6eb14983ab7eaf38ff138c3c397a6fe3b/pandas/_libs/src/ujson/lib/ultrajsondec.c#L128
But I'm down to rewrite this part to follow the PyLong_AsLongLongAndOverflow implementation:
```
https://github.com/python/cpython/blob/04cdeb7a5617c48102f45b965e683b12cdf934f8/Objects/longobject.c#L380
```

Reply:
The problem is if `intValue*10ULL + newDigit` overflows than the comparison to `overflowLimit` is a moot point because you've already hit undefined behavior with the former